libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayEvents.kt

```diff
@@ -0,0 +1,34 @@
+package com.mapbox.navigation.core.replay.history
+
+import com.google.gson.annotations.SerializedName
+
+data class ReplayEvents(
+    // Assumes chronological order, index 0 moves to events.size over time.
+    val events: List<ReplayEventBase>
+)
+
+abstract class ReplayEventBase(
```

Comment:
yeah same.. i tried interface it failed to use `@SerializedName("event_timestamp")`
![Screen Shot 2020-04-03 at 3 14 42 PM](https://user-images.githubusercontent.com/3021882/78409208-f05b3780-75bd-11ea-9693-0cefc05e57cd.png)

the `@SerializedName` in an interface worked on a `fun eventTimestamp()`

and then it looks like this. I also don't like @SerializedName on functions because what does that even mean?
```
interface ReplayEventBase {
    @SerializedName("event_timestamp")
    fun eventTimestamp(): Double
}

data class ReplayEventGetStatus(
    val eventTimestamp: Double,
    val timestamp: Double
) : ReplayEventBase {
    override fun eventTimestamp(): Double {
        return eventTimestamp
    }
}
```

Reply:
That's what I meant
> That val is not a field is a function, interfaces cannot have fields, that gets compiled to `getEventTimestamp` and hence the error.

> and then it looks like this. I also don't like @SerializedName on functions because what does that even mean?

👀 the diff adding the `@SerializedName` to the `override` works 👌 a bit of boilerplate but ¯\\\_(ツ)\_/¯

Reply:
ah yeah a little less boilerplate. taking it

 data class ReplayEventGetStatus(
    `@SerializedName("event_timestamp")`
     override val eventTimestamp: Double,
     val timestamp: Double
) : ReplayEventBase